graphql_compiler/query_pagination/pagination_planning.py

```diff
@@ -1,28 +1,38 @@
 # Copyright 2019-present Kensho Technologies, LLC.
-from collections import namedtuple
+from typing import NamedTuple, Tuple
 
 from ..ast_manipulation import get_only_query_definition, get_only_selection_from_ast
-from ..cost_estimation.helpers import is_int_field_type, is_uuid4_type
-from ..exceptions import GraphQLError
+from ..cost_estimation.helpers import is_uuid4_type
+from ..cost_estimation.int_value_conversion import field_supports_range_reasoning
+from ..exceptions import GraphQLError, GraphQLPaginationError
 
 
-# The intent to split the query at a certain vertex into a certain number of pages.
-VertexPartition = namedtuple(
-    "VertexPartition",
-    (
-        "query_path",  # Tuple[field name : str] leading to the vertex to be split
-        "number_of_splits",  # The number of subdivisions intended for this vertex
-    ),
-)
+class VertexPartition(NamedTuple):
+    """The intent to split the query at a certain vertex into a certain number of pages."""
 
+    # field names leading to the vertex to be split
+    query_path: Tuple[str, ...]
 
-# The intent to split the query with a combination of VertexPartitions
-PaginationPlan = namedtuple("PaginationPlan", ("vertex_partitions",))  # List[VertexPartition]
+    # field to use for pagination
+    pagination_field: str
+
+    # The number of subdivisions intended for this vertex
+    number_of_splits: int
+
+
+class PaginationPlan(NamedTuple):
+    """The intent to split the query with a combination of VertexPartitions"""
+
+    vertex_partitions: Tuple[VertexPartition, ...]
+
+
+# A field with N quantiles has a pagination capacity N / <QUANTILE_REQUIREMENT_FACTOR>
+QUANTILE_REQUIREMENT_FACTOR = 5
```

Comment:
Could you describe this in a bit more detail? I'm not 100% sure what the intent here is and what problem is being solved this way.

I'd also remove the angle-brackets from `<QUANTILE_REQUIREMENT_FACTOR>`, I think that might improve readability.